### PR TITLE
Feature/revamp quiz and minor scoring issues

### PR DESCRIPTION
**Pull request by DavidWilliamK**
- Revamp all quiz related page
- Changes to landing page
- Updated toast to use buefy
- Added emptyState for scoring-related pages
- Added report page to see the total points of all students from a batch
- Added copy feature
- Implemented timer and router guard feature at quiz
- Unit test

### CODE REVIEW

File: src/api/default-request.js [hunk 1]
```diff
@@ -21,6 +21,8 @@ const getRequest = function (path, callback, errorHandler, configuration) {
       callback(data)
     })
     .catch(error => {
+      console.log(path)
```
Comment (oliverrsebastian) [inline src/api/default-request.js]: is this expected?

File: src/config/index.js [hunk 2]
```diff
@@ -295,10 +296,10 @@ module.exports = {
           },
           comments: {
             list(batchCode, assignmentId, studentId, page, pageSize) {
-              return `/api/scoring/batches/${batchCode}/assignments/${assignmentId}/room/${studentId}/comments?page=${page}&size=${pageSize}`
+              return `/api/scoring/batches/${batchCode}/assignments/${assignmentId}/rooms/${studentId}/comments?page=${page}&size=${pageSize}`
```
Comment (oliverrsebastian) [inline src/config/index.js]: why is this changed? it should be `room` instead of `rooms`

File: src/config/index.js [hunk 3]
```diff
@@ -358,8 +359,8 @@ module.exports = {
           return `/api/scoring/batches/${batchCode}/quizzes/${id}`
         },
         students: {
-          list(batchCode, page, pageSize) {
-            return `/api/scoring/batches/${batchCode}/quizzes?page=${page}&size=${pageSize}`
+          list(batchCode, isPassedDeadline, page, pageSize) {
```
Comment (oliverrsebastian) [inline src/config/index.js]: wanted to differentiate student api and quiz list api even though the value is identical?

File: src/views/QuestionBank/js/question-bank-detail.js [hunk 4]
```diff
@@ -87,6 +102,9 @@ export default {
       }
     },
     failFetchingQuestionBankQuestionList () {
+      this.isLoading = false
+      this.failLoadQuestion = true
+      //TODO : CHANGE TO EMPTY STATE
```
Comment (oliverrsebastian) [inline src/views/QuestionBank/js/question-bank-detail.js]: already working this TODO for the next development?

Comment (DavidWilliamK) [reply src/views/QuestionBank/js/question-bank-detail.js]: Already implemented, forgot to remove TODO

File: src/api/default-request.js (hunk 1 above)
Comment (DavidWilliamK) [reply src/api/default-request.js]: No, forgot to remove this after tracking error.
Will remove

File: src/config/index.js (hunk 3 above)
Comment (DavidWilliamK) [reply src/config/index.js]: Yes, intended for readability on the studentQuiz list's javascript part.
However not sure if this is the best practice.
